WalletWasabi/Crypto/Mac.cs

```diff
@@ -19,17 +19,6 @@ private MAC(Scalar t, GroupElement v)
 		public GroupElement V { get; }
 		internal GroupElement U => GenerateU(T);
 
-		public static bool operator ==(MAC? a, MAC? b) => (a?.T, a?.V) == (b?.T, b?.V);
-
-		public static bool operator !=(MAC? a, MAC? b) => !(a == b);
-
-		public override bool Equals(object? obj) => Equals(obj as MAC);
-
-		public bool Equals(MAC? other) => this == other;
-
-		public override int GetHashCode() =>
-			HashCode.Combine(T, V).GetHashCode();
```

Comment:
I'd prefer to not lose this functionality.

Reply:
That functionality is still there because now the compiler implements a bugs-free version by us for free, that's the beauty of records. That's why all the tests still pass even when the heavily depend on the `IEquatable<MAC>`.